### ISSUE TITLE
Wrong calculation when negative energy price

### ISSUE THREAD

**Issue by maskedfighter** [mampfes/ha_epex_spot#34]
Hi,

When energy prices drop below 0, tax and additional fees are calculated on top, which would mean that when i pay 20% tax at -500€/Mwh i would get 600€/Mwh but actually it should be 400€ right? So there seems to be something abnor al when prices are below 0, calculation is wrong then.

**Comment by mampfes** [comment]
Oh, yes - I think you are right.I will fix this.

**Comment by hupfis** [comment]
I actually believe that the 600€/Mwh is the correct price. @mampfes before doing a change here, shouldn't we check back with any of the companies to understand tha actual calculation. 
My understanding with random numbers for a random month
I have positive prices of 40 € (net)
I have negative prices of -10 € (net)
I will have to pay for 30 € (net), given AT (20% tax), I will be charged for a total of €36

That equals the gross price of 
* positive price: 48 € (gross)
* negative price: -12 € (gross)
----------------------------
equals 36 €  (gross)

**Comment by maskedfighter** [comment]
I understand your point but in that one hour the price was at 500€/MWh i would've gotten more money then awattar acutally calculated, this would assume that i would get 20% on top instead of paying it.

**Comment by maskedfighter** [comment]
Ok you might have a point there, awattar calculates it also the way it is implemented right now. Then it's just me not understanding the logic behind it..

**Comment by hupfis** [comment]
If the NET price is at +300€, you pay 360€
If the NET price is at -300€, you "get" 360€
I put "get" in quotes as the billing will only sum up the NET prices and then add the tax. 
Therefore, if you want to break down the actual "gross" earnings you can assume adding the tax also to the negative prices by multiplying them with 1.2 (for Austrian 20% tax)

(Keep in mind my opinion here is based on my understanding of business+rules in Austria, they might be wrong as I am not an accountant by profession)

**Comment by maskedfighter** [comment]
You might be right since awattar does it the same way, at least from what i can see from the diagram on their website.

**Comment by mampfes** [comment]
Ok, before I do any changes, we should definitely know the companies are handling the various surcharges. I am not a customer of Awattar/Tibber, etc. Therefore I don't have access to bills... On their website I didn't find a clear description.

I can imagine that e.g. Awattar is not paying the 3% surcharge for negative market prices. And I have not glue how they handle the tax for a negative price.

**Comment by gerione** [comment]
Awattar is taking the absolute value from the market price.  
![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/2fbc48d1-fb53-4a6a-a4b2-d04585c1714f)
So if my price is -10 cents/kWh. The 3% surcharge is calculated from the abs(-10cents/kWh)*0.03 -> 3cents. + 1.5 cents
The 20% Tax is taken from the -10cents -> -2 cents. 
-10 + 3 + 1.5 - 2 cents = -7.5cents/kWh
You can see this from the Netto/Brutto chart they are providing
![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/ec14faed-1af3-4315-a5ec-911e15e0c731)

![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/6276dbec-27d8-40cb-ae1c-d63f14e652f3)

**Comment by mampfes** [comment]
Ok, then the following calculation should be correct, right?
```
        net_p = price_eur_per_mwh / 10  # convert from EUR/MWh to ct/kWh
        net_p = net_p + abs(net_p * surcharge_pct / 100)
        net_p += surcharge_abs
        net_p *= 1 + (tax / 100)
```

Previously, the net price was calculated as follows:
```
        net_p = price_eur_per_mwh / 10  # convert from EUR/MWh to ct/kWh
        net_p *= 1 + ( surcharge_pct / 100)
        net_p += surcharge_abs
        net_p *= 1 + (tax / 100)
```

**Comment by mampfes** [comment]
fixed by 6337b4e
